### PR TITLE
feat: IdP Management + M2M Auth for Databricks

### PR DESCRIPTION
**Pull request by chrisbonilla95**
Enable management of IdPs via SDK as well as add support for configuring M2M authentication for Databricks data source type.

JIRA: LX-451, LX-404, LX-617
risk: low

### CODE REVIEW

File: gooddata-sdk/gooddata_sdk/catalog/organization/service.py [hunk 1]
```diff
@@ -324,6 +327,95 @@ def update_csp_directive(self, csp_directive: CatalogCspDirective) -> None:
...
+        Raises:
+            ValueError:
+                Identity provider does not exist.
+        """
+        try:
+            self._entities_api.delete_entity_identity_providers(identity_provider_id)
+        except NotFoundException:
+            raise ValueError(
+                f"Can not delete {identity_provider_id} identity provider. " f"This identity provider does not exist."
+            )
```
Comment (hkad98) [inline gooddata-sdk/gooddata_sdk/catalog/organization/service.py:395]: Not sure if try-except is necessary for deletion. If it is implemented as majority of APIs it should no fail if `identity_provider_id` does not exist. Could you please verify if it is true what I am saying?

Comment (chrisbonilla95) [reply gooddata-sdk/gooddata_sdk/catalog/organization/service.py:395]: Looks like it does throw `NotFoundException`

```
 if r.status == 404:
>               raise NotFoundException(http_resp=r)
E               gooddata_api_client.exceptions.NotFoundException: Status Code: 404
E               Reason: Not Found
E               HTTP response headers: HTTPHeaderDict({'Access-Control-Allow-Credentials': 'true', 'Access-Control-Expose-Headers': 'Content-Disposition, Content-Length, Content-Range, Set-Cookie', 'Cache-Control': 'no-cache, no-store, max-age=0, must-revalidate', 'Connection': 'keep-alive', 'Content-Length': '172', 'Content-Security-Policy': "default-src 'self' *.wistia.com *.wistia.net; script-src 'self' 'unsafe-inline' 'unsafe-eval' *.wistia.com *.wistia.net *.hsforms.net *.hsforms.com src.litix.io matomo.anywhere.gooddata.com *.jquery.com unpkg.com cdnjs.cloudflare.com; img-src * data: blob:; style-src 'self' 'unsafe-inline' fonts.googleapis.com cdn.jsdelivr.net fast.fonts.net; font-src 'self' data: fonts.gstatic.com *.alicdn.com *.wistia.com cdn.jsdelivr.net info.gooddata.com; frame-src 'self' *.hsforms.net *.hsforms.com; object-src 'none'; worker-src 'self' blob:; child-src blob:; connect-src 'self' *.tiles.mapbox.com *.mapbox.com *.litix.io *.wistia.com *.hsforms.net *.hsforms.com embedwistia-a.akamaihd.net matomo.anywhere.gooddata.com; media-src 'self' blob: data: *.wistia.com *.wistia.net embedwistia-a.akamaihd.net", 'Content-Type': 'application/problem+json', 'DATE': 'PLACEHOLDER', 'Expires': '0', 'GoodData-Deployment': 'aio', 'Permission-Policy': "geolocation 'none'; midi 'none'; sync-xhr 'none'; microphone 'none'; camera 'none'; magnetometer 'none'; gyroscope 'none'; fullscreen 'none'; payment 'none';", 'Pragma': 'no-cache', 'Referrer-Policy': 'no-referrer', 'Server': 'nginx', 'Vary': 'Origin, Access-Control-Request-Method, Access-Control-Request-Headers', 'X-Content-Type-Options': 'nosniff', 'X-GDC-TRACE-ID': 'PLACEHOLDER', 'X-XSS-Protection': '0', 'set-cookie': 'SPRING_REDIRECT_URI=; Max-Age=0; Expires=Mon, 25 Nov 2024 14:59:24 GMT; Path=/; HTTPOnly; SameSite=Lax'})
E               HTTP response body: {"detail": "The requested endpoint does not exist or you do not have permission to access it.", "status": 404, "title": "Not Found", "traceId": "674490cc6c7b69e73b0d3e3e900238a4"}

../gooddata-api-client/gooddata_api_client/rest.py:225: NotFoundException
```

Comment (hkad98) [reply gooddata-sdk/gooddata_sdk/catalog/organization/service.py:395]: I tried it and it does not raise NotFoundException if it does not exist.

```
>>> sdk = GoodDataSdk.create_from_profile("panther")
>>> sdk.support.is_available
True
>>> sdk.client.entities_api.delete_entity_identity_providers("i-do-not-exist")
>>> 
```

Comment (chrisbonilla95) [reply gooddata-sdk/gooddata_sdk/catalog/organization/service.py]: Ah, you're right. I tested by modifying the existing test for delete and was getting error but it was from the get that proceeded.

**Review by hkad98 — APPROVED** [review]

**Pull request merged**